task_2/join.go

```diff
@@ -0,0 +1,124 @@
+package task_2
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func Start() {
+
+	var (
+		numOfChannels = rand.Intn(12) + 1
+		channels      []chan interface{}
+	)
+
+	fmt.Println(fmt.Sprintf("creating %d channels..", numOfChannels))
+	for i := 0; i < numOfChannels; i++ {
+		channels = append(channels, make(chan interface{}))
+	}
+
+	ctx, _ := context.WithTimeout(context.Background(), 35*time.Second)
+
+	joinedChan := join(ctx, channels...)
+	go writeToChannels(channels)
+	for {
+		select {
+		case <-time.After(10 * time.Second):
+			fmt.Println("reached time limit waiting for joined output, exiting.")
+			return
+		case joinedOutput, ok := <-joinedChan:
+			if !ok {
+				return
+			}
+			fmt.Println("[RECEIVED JOINED OUTPUT]: ", joinedOutput)
+		}
+	}
+
+}
+
+func writeToChannels(ch []chan interface{}) {
+	for i := 1; i < 10; i++ {
+		fmt.Println(fmt.Sprintf("-- CYCLE %d -- sending messages to channels..", i))
+		for j, c := range ch {
```

Comment:
It is not fair enough :) 
Every channel you trying to send data into is an unbuffered channel. That is why you will wait until the message will be read from the receiver side. Also, you guarantee that no new messages will be sent to the same channel until the receiver side reads the previous one. That leads us to a more predictable result. 

Maybe this code snipper can help you to understand what I actually mean:
```go
func newSequence(ctx context.Context) <-chan int {
	var out = make(chan int)
	go func() {
		for {
			select {
			case <-ctx.Done():
				return
			case <-time.After(time.Duration(rand.Intn(1000)) * time.Millisecond):
				select {
				case <-ctx.Done():
					return
				case out <- rand.Intn(1000):
				}
			}
		}
	}()
	return out
}

func main() {
        ...
        ctx := context.Background()
        ...
	s1 := newSequence(ctx)
	s2 := newSequence(ctx)
        ...
}
```